packages/core/ui-settings/core-ui-settings-server-internal/src/ui_settings_service.ts

```diff
@@ -87,36 +89,52 @@ export class UiSettingsService
     this.validatesOverrides();
 
     return {
-      asScopedToClient: this.getScopedClientFactory(),
+      asScopedToClient: this.getScopedClientFactory('namespace'),
+      asScopedToGlobalClient: this.getScopedClientFactory('global'),
     };
   }
 
   public async stop() {}
 
-  private getScopedClientFactory(): (
-    savedObjectsClient: SavedObjectsClientContract
-  ) => UiSettingsClient {
+  private getScopedClientFactory(
+    scope: UiSettingsScope
+  ): (savedObjectsClient: SavedObjectsClientContract) => UiSettingsClient | UiSettingsGlobalClient {
     const { version, buildNum } = this.coreContext.env.packageInfo;
-    return (savedObjectsClient: SavedObjectsClientContract) =>
-      new UiSettingsClient({
-        type: 'config',
+    return (savedObjectsClient: SavedObjectsClientContract) => {
+      const isNamespaceScope = scope === 'namespace';
+
+      const options = {
+        type: (isNamespaceScope ? 'config' : 'config-global') as 'config' | 'config-global',
         id: version,
         buildNum,
         savedObjectsClient,
-        defaults: mapToObject(this.uiSettingsDefaults),
-        overrides: this.overrides,
+        defaults: isNamespaceScope
+          ? mapToObject(this.uiSettingsDefaults)
+          : mapToObject(this.uiSettingsGlobalDefaults),
+        overrides: isNamespaceScope ? this.overrides : {},
         log: this.log,
-      });
+      };
+      return UiSettingsClientFactory.create(options);
+    };
   }
 
-  private register(settings: Record<string, UiSettingsParams> = {}) {
+  private register = (settings: Record<string, UiSettingsParams> = {}) => {
```

Comment:
no reason, I prefer arrow functions in general

Reply:
You know this is plain worse in term of performance though, right? As arrow functions as class properties are per-object instead of being provided by the prototype? Unless strictly necessary (e.g passing the function reference as a handler without binding it) it should be avoided.